app/models/manageiq/providers/openstack/infra_manager/orchestration_stack.rb

```diff
@@ -138,5 +160,17 @@ def log_and_raise_update_error(method_name, err)
     raise MiqException::MiqOrchestrationUpdateError, err.to_s, err.backtrace
   end
 
+  def workflow_service
+    ext_management_system.workflow_service
```

Comment:
Should this be `ext_management_system.openstack_handle.workflow_service` or (or `detect_workflow_service` to handle the exception)?

Reply:
Nevermind, I see that `ext_management_system.workflow_service` is actually calling `ext_management_system.openstack_handle.detect_workflow_service` on the InfraManager.